### PR TITLE
feat: Add HavePrivateParameterlessConstructor and NotHavePrivateParameterlessConstructor

### PR DESCRIPTION
**Pull request by Redestros**
## Description

This PR adds support for checking if classes have private parameterless constructors in ArchUnitNET. This feature is particularly valuable for enforcing **Domain-Driven Design** patterns where entities should have private parameterless constructors for ORM frameworks while maintaining public constructors with parameters for domain logic.

## Motivation

In Domain-Driven Design and Entity Framework scenarios, it's common to require:
- **Private parameterless constructors** for ORM frameworks (Entity Framework, NHibernate, etc.)
- **Public parameterized constructors** for domain logic and business rules
- **Abstract classes excluded** from constructor requirements

This pattern ensures that:
- ORM frameworks can instantiate entities during deserialization
- Domain logic enforces proper entity creation through meaningful constructors
- Encapsulation and invariants are maintained

### Core Changes
- ✅ **Added `HavePrivateParameterlessConstructor()` condition** using `SimpleCondition<Class>`
- ✅ **Added `NotHavePrivateParameterlessConstructor()` condition** for inverse validation
- ✅ **Integrated into fluent API** via `IClassConditions` interface and `ClassConditionsDefinition`
- ✅ **Proper nullable handling** for `IsAbstract` property (`cls.IsAbstract == true`)
- ✅ **Abstract class exclusion** - automatically satisfy both conditions

### Test Coverage
- ✅ **Comprehensive unit tests** covering all scenarios
- ✅ **Edge case handling** (abstract classes, parameterized-only constructors)

### Basic Usage
```csharp
// Ensure domain entities have private parameterless constructors
var rule = Classes()
    .That().ResideInNamespace("MyApp.Domain.Entities")
    .And().AreNotAbstract()
    .Should().HavePrivateParameterlessConstructor()
    .Because("Domain entities need private constructors for ORM frameworks");

rule.Check(architecture);

### CODE REVIEW

**Comment by Redestros** [comment]
Hi @alexanderlinne ,
Thank you for the feedback! You're absolutely right - the compositional approach is much more flexible. I've updated the PR accordingly.
Changes Made
I've removed the specific HavePrivateParameterlessConstructor condition and instead implemented the granular HaveAnyParameters and NotHaveAnyParameters conditions as you suggested.
New Implementation:

✅ Added HaveAnyParameters() condition for MethodMembers
✅ Added NotHaveAnyParameters() condition for MethodMembers
✅ Added method signatures to IMethodMemberConditions interface
✅ Implemented in MethodMemberConditionsDefinition fluent API
✅ Added comprehensive test coverage

Usage Example
Your suggested syntax now works perfectly:
```csharp
var classes = Classes()
    .That().ResideInNamespace("MyApp.Domain.Entities")
    .And().AreNotAbstract();

MethodMembers()
    .That()
    .AreDeclaredIn(classes)
    .And()
    .AreConstructors()
    .Should()
    .BePrivate()
    .AndShould()
    .NotHaveAnyParameters();

File: ArchUnitNET/Fluent/Syntax/Elements/Members/MethodMembers/MethodMembersShould.cs [hunk 1]
```diff
@@ -11,21 +15,6 @@ public class MethodMembersShould
         public MethodMembersShould(IArchRuleCreator<MethodMember> ruleCreator)
             : base(ruleCreator) { }
 
-        public ShouldRelateToMethodMembersThat<
```
Comment (alexanderlinne) [inline ArchUnitNET/Fluent/Syntax/Elements/Members/MethodMembers/MethodMembersShould.cs]: This function should not be removed

**Review by alexanderlinne — CHANGES_REQUESTED** [review]
There are several unexpected changes in your formatting commit. Please just use csharpier for that
